### ISSUE TITLE
Unresolved Reference for com.ibm.uvm.tools in an OSGI Bundle

### ISSUE THREAD

**Issue by dsearles-celerium** [ESAPI/esapi-java-legacy#617]
When I add **esapi-2.2.3.0.jar** to my OSGI project, the bundle will not load due to an unresolved reference to **com.ibm.uvm.tools**. I've searched high and low for an artifact named **com.ibm.uvm.tools** without success. I realize this is not a direct reference from ESAPI itself but the addition of this one JAR (**esapi-2.2.3.0.jar**) causes the unresolved reference which prevents my OSGI bundle from loading. I'm hoping someone may have run into this issue before and can help me identify a workaround. As it stands now, I cannot use the ESAPI JAR due to this issue.

**Comment by xeno6696** [comment]
@dsearles-celerium this is something unique to your environment.  

```
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------------< org.owasp.esapi:esapi >------------------------
[INFO] Building ESAPI 2.2.4.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:3.1.2:list (default-cli) @ esapi ---
[INFO] 
[INFO] The following files have been resolved:
[INFO]    javax.servlet:javax.servlet-api:jar:3.0.1:provided -- module javax.servlet.api (auto)
[INFO]    javax.servlet.jsp:javax.servlet.jsp-api:jar:2.3.3:provided -- module javax.servlet.jsp.api (auto)
[INFO]    com.io7m.xom:xom:jar:1.2.10:compile -- module xom (auto)
[INFO]    commons-beanutils:commons-beanutils:jar:1.9.4:compile -- module commons.beanutils (auto)
[INFO]    commons-logging:commons-logging:jar:1.2:compile -- module commons.logging (auto)
[INFO]    commons-collections:commons-collections:jar:3.2.2:compile -- module commons.collections (auto)
[INFO]    commons-configuration:commons-configuration:jar:1.10:compile -- module commons.configuration (auto)
[INFO]    commons-lang:commons-lang:jar:2.6:compile -- module commons.lang (auto)
[INFO]    commons-fileupload:commons-fileupload:jar:1.3.3:compile -- module commons.fileupload (auto)
[INFO]    log4j:log4j:jar:1.2.17:compile -- module log4j (auto)
[INFO]    org.apache.commons:commons-collections4:jar:4.2:compile -- module org.apache.commons.collections4 [auto]
[INFO]    org.apache-extras.beanshell:bsh:jar:2.0b6:compile -- module bsh (auto)
[INFO]    org.owasp.antisamy:antisamy:jar:1.6.2:compile -- module antisamy (auto)
[INFO]    net.sourceforge.nekohtml:nekohtml:jar:1.9.22:compile -- module nekohtml (auto)
[INFO]    org.apache.httpcomponents:httpclient:jar:4.5.13:compile -- module org.apache.httpcomponents.httpclient [auto]
[INFO]    org.apache.httpcomponents:httpcore:jar:4.4.14:compile -- module org.apache.httpcomponents.httpcore [auto]
[INFO]    org.apache.xmlgraphics:batik-css:jar:1.14:compile -- module batik.css (auto)
[INFO]    org.apache.xmlgraphics:batik-shared-resources:jar:1.14:compile -- module batik.shared.resources (auto)
[INFO]    org.apache.xmlgraphics:batik-util:jar:1.14:compile -- module batik.util (auto)
[INFO]    org.apache.xmlgraphics:batik-constants:jar:1.14:compile -- module batik.constants (auto)
[INFO]    org.apache.xmlgraphics:batik-i18n:jar:1.14:compile -- module batik.i18n (auto)
[INFO]    org.apache.xmlgraphics:xmlgraphics-commons:jar:2.6:compile -- module xmlgraphics.commons (auto)
[INFO]    commons-io:commons-io:jar:1.3.1:compile -- module commons.io (auto)
[INFO]    org.slf4j:slf4j-simple:jar:1.7.30:compile -- module org.slf4j.simple [auto]
[INFO]    xerces:xercesImpl:jar:2.12.1:compile -- module xercesImpl (auto)
[INFO]    xml-apis:xml-apis-ext:jar:1.3.04:compile -- module xml.apis.ext (auto)
[INFO]    org.slf4j:slf4j-api:jar:1.7.30:compile -- module org.slf4j [auto]
[INFO]    xml-apis:xml-apis:jar:1.4.01:compile -- module xml.apis (auto)
[INFO]    com.github.spotbugs:spotbugs-annotations:jar:4.2.2:compile (optional)  -- module com.github.spotbugs.annotations [auto]
[INFO]    com.google.code.findbugs:jsr305:jar:3.0.2:compile (optional)  -- module jsr305 (auto)
[INFO]    commons-codec:commons-codec:jar:1.15:test -- module org.apache.commons.codec [auto]
[INFO]    junit:junit:jar:4.13.2:test -- module junit [auto]
[INFO]    org.bouncycastle:bcprov-jdk15on:jar:1.68:test -- module org.bouncycastle.provider
[INFO]    org.hamcrest:hamcrest-core:jar:1.3:test -- module hamcrest.core (auto)
[INFO]    org.powermock:powermock-api-mockito2:jar:2.0.7:test -- module powermock.api.mockito2 (auto)
[INFO]    org.powermock:powermock-api-support:jar:2.0.7:test -- module powermock.api.support (auto)
[INFO]    org.javassist:javassist:jar:3.25.0-GA:test -- module javassist (auto)
[INFO]    org.mockito:mockito-core:jar:2.28.2:test -- module org.mockito [auto]
[INFO]    net.bytebuddy:byte-buddy:jar:1.9.10:test -- module net.bytebuddy
[INFO]    net.bytebuddy:byte-buddy-agent:jar:1.9.10:test -- module net.bytebuddy.agent
[INFO]    org.objenesis:objenesis:jar:2.6:test -- module objenesis (auto)
[INFO]    org.powermock:powermock-core:jar:2.0.7:test -- module powermock.core (auto)
[INFO]    org.powermock:powermock-module-junit4:jar:2.0.7:test -- module powermock.module.junit4 (auto)
[INFO]    org.powermock:powermock-module-junit4-common:jar:2.0.7:test -- module powermock.module.junit4.common (auto)
[INFO]    org.powermock:powermock-reflect:jar:2.0.7:test -- module powermock.reflect (auto)
[INFO]    org.openjdk.jmh:jmh-core:jar:1.28:test -- module jmh.core (auto)
[INFO]    net.sf.jopt-simple:jopt-simple:jar:4.6:test -- module jopt.simple (auto)
[INFO]    org.apache.commons:commons-math3:jar:3.2:test -- module commons.math3 (auto)
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.405 s
[INFO] Finished at: 2021-03-30T11:40:48-07:00
[INFO] ------------------------------------------------------------------------

```
After blowing away my .m2 directory and running dependency:list after a standard "mvn clean test" I don't have an "IBM" _anything_ in my .m2 directory.

**Comment by kwwall** [comment]
@xeno6696 - As I mentioned in https://github.com/ESAPI/esapi-java/issues/12#issuecomment-810434953, we do have a "com.ibm" package name reference for the IBMJCE JCE Provider, but that is the only reference.

@dsearles-celerium - Did you have this working with an earlier version of ESAPI or is this your first attempt to get it working? Again, I don't mind assisting you in troubleshooting to help you get ESAPI working, but the ESAPI-Users mailing list is a much more appropriate forum for that than here.) ESAPI has no references to any other com.ibm package except for the JCE provider and certainly not to com.ibm.uvm.tools. (I suppose that their is a transitive dependency that could get pulled in by reflection though that wouldn't show up at compile time.) But this sounds more like something specifically related to OSGI and maybe your container. (Are you using IBM WebSphere perhaps?) E.g., see https://stackoverflow.com/questions/13585513/cannot-start-osgi-bundle-missing-com-ibm-uvm-tools

**Comment by xeno6696** [comment]
@dsearles-celerium This stackoverflow question looks like it might be directly relevant as there were some logging changes especially if you haven't updated inawhile:

https://stackoverflow.com/a/13934123/557153

> Solution was to use a OSGified version of log4j as provided dependency. I had it as a compiled dependency. Log4j manifest mentions this package as an optional depndency, so maven-bundle-plugin added it to my bundle's manifest, but did not make it optional.

**Comment by dsearles-celerium** [comment]
@xemo6696 Yes, I had seen that stackoverflow post but it didn't seem to be applicable. 
@kwwall This is in a Liferay portal and I just added the ESAPI JAR in order to use ESAPI.httpUtilities().addCookie(HttpServletResponse,Cookie) to resolve some "flaws" found by a pen test. I also suspect it could be some downstream dependency in some artifact used by ESAPI.

Since I'm under the gun timewise and was only using addCookie(), for now I just found the implementation of that method and associated regex's and implemented my own version for now. When the dust settles, I'll try to bring in ESAPI again.

Thanks to both of your for your looking at this!
